jlibra-core/src/main/java/dev/jlibra/serialization/lcs/LCSSerializer.java

```diff
@@ -0,0 +1,68 @@
+package dev.jlibra.serialization.lcs;
+
+import static java.util.stream.Collectors.toList;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.stream.Stream;
+
+import dev.jlibra.LibraRuntimeException;
+import dev.jlibra.serialization.ByteSequence;
+import dev.jlibra.serialization.Serializer;
+
+public class LCSSerializer {
+
+    public ByteSequence serialize(Object serializable, Class<?> type) {
+        Serializer s = Serializer.builder();
+
+        LCS.ExternallyTaggedEnumeration enumAnnotation = type.getAnnotation(LCS.ExternallyTaggedEnumeration.class);
+        if (enumAnnotation != null) {
+            s = s.appendInt(enumAnnotation.value());
+        }
+
+        List<Method> methods = Stream.of(type.getMethods())
+                .filter(m -> m.getAnnotation(LCS.Field.class) != null)
+                .sorted((m1, m2) -> m1.getDeclaredAnnotation(LCS.Field.class).value()
+                        - m2.getDeclaredAnnotation(LCS.Field.class).value())
+                .collect(toList());
+
+        for (Method m : methods) {
+            Class<?> returnType = m.getReturnType();
+            if (returnType.getAnnotation(LCS.Structure.class) != null
+                    || returnType.getAnnotation(LCS.ExternallyTaggedEnumeration.class) != null) {
+                Object l = invokeMethod(serializable, m);
+                ByteSequence value = serialize(l, returnType);
+                s = s.appendFixedLength(value);
```

Comment:
What if you return here instead of using `s` var? You could get rid of if-else piramid. Not sure if that's nicer. Just an idea.